### PR TITLE
[Images] [7/N] Add image encoding support.

### PR DESCRIPTION
**Pull request by clarkzinzow**
This PR adds support for encoding and image column into a binary column, containing bytes encoded into the specified file format.

This serves as a precursor to both more user-friendly reprs in Jupyer Notebooks and saving/uploading images to specified URIs.

### CODE REVIEW

File: daft/expressions/expressions.py [hunk 1]
```diff
@@ -668,10 +669,45 @@ def resolve_schema(self, schema: Schema) -> Schema:
...
+        """
+        Resize image into the provided width and height.
+
+        Args:
+            w: Desired width of the resized image.
+            h: Desired height of the resized image.
+
+        Returns:
+            Expression: An Image expression representing an image column of the resized images.
+        """
```
Comment (clarkzinzow) [inline daft/expressions/expressions.py:710]: @samster25 Also added some missing docstrings for image expression ops as a driveby!

Comment (samster25) [reply daft/expressions/expressions.py:710]: Thanks!

File: daft/series.py [hunk 2]
```diff
@@ -468,6 +468,11 @@ class SeriesImageNamespace(SeriesNamespace):
     def decode(self) -> Series:
         return Series._from_pyseries(self._series.image_decode())
 
+    def encode(self, image_format: str | ImageFormat) -> Series:
+        if isinstance(image_format, str):
+            image_format = ImageFormat.from_format_string(image_format.upper())
```
Comment (samster25) [inline daft/series.py:473]: Have an else case here if image_format isn't a str or ImageFormat. Otherwise we panic on the type extract in rust

File: src/array/ops/image.rs [hunk 3]
```diff
@@ -371,6 +398,25 @@ impl ImageArray {
         Some(result)
     }
 
+    pub fn encode(&self, image_format: ImageFormat) -> DaftResult<BinaryArray> {
+        let result = (0..self.len())
+            .map(|i| self.as_image_obj(i))
+            .map(|img| {
+                img.map(|img| {
+                    let mut buf = Vec::new();
+                    img.encode(image_format, &mut buf)?;
```
Comment (samster25) [inline src/array/ops/image.rs:407]: I should have clarified better but I was thinking we have the `img.encode` extend single buffer that is passed in rather than allocate a new buffer for each image and then concat it after
